projects/packages/jetpack-mu-wpcom/src/features/launchpad/class-launchpad-task-lists.php

```diff
@@ -310,6 +311,22 @@ private function load_value_from_callback( $item, $callback, $default = '' ) {
 		return $default;
 	}
 
+	/**
+	 * Loads any extra data for a task, calling the 'extra_data_callback' callback if it exists,
+	 * or an empty array if there is no callback.
+	 *
+	 * @param Task $task A task definition.
+	 * @return array The extra data for the task.
+	 */
+	private function load_extra_data( $task ) {
+		$extra_data = $this->load_value_from_callback( $task, 'extra_data_callback' );
+		if ( ! empty( $extra_data ) ) {
+			return $extra_data;
+		}
+
+		return array();
```

Comment:
As discussed above, I think we should update the return type to be `array|null` so we can avoid returning unnecessary data over the wire. We should also enforce that we're actually getting an array back.
```suggestion
	 * @return array|null The extra data for the task.
	 */
	private function load_extra_data( $task ) {
		$extra_data = $this->load_value_from_callback( $task, 'extra_data_callback' );
		if ( is_array( $extra_data ) && array() !== $extra_data ) {
			return $extra_data;
		}

		return null;
```